lnd_multihopreceiverchainclaim_test.go

```diff
@@ -22,52 +24,63 @@ import (
 func testMultiHopReceiverChainClaim(net *lntest.NetworkHarness, t *harnessTest) {
 	ctxb := context.Background()
 
-	// First, we'll create a three hop network: Alice -> Bob -> Carol, with
-	// Carol refusing to actually settle or directly cancel any HTLC's
-	// self.
+	// First, we'll create a three hop network: Alice -> Bob -> Carol.
 	aliceChanPoint, bobChanPoint, carol :=
-		createThreeHopNetwork(t, net, true)
+		createThreeHopNetwork(t, net, false)
 
 	// Clean up carol's node when the test finishes.
 	defer shutdownAndAssert(net, t, carol)
 
-	// With the network active, we'll now add a new invoice at Carol's end.
-	// Make sure the cltv expiry delta is large enough, otherwise Bob won't
-	// send out the outgoing htlc.
+	// With the network active, we'll now add a new hodl invoice at Carol's
+	// end. Make sure the cltv expiry delta is large enough, otherwise Bob
+	// won't send out the outgoing htlc.
+
 	const invoiceAmt = 100000
-	invoiceReq := &lnrpc.Invoice{
+	preimage := lntypes.Preimage{1, 2, 4}
+	payHash := preimage.Hash()
+	invoiceReq := &invoicesrpc.AddHoldInvoiceRequest{
 		Value:      invoiceAmt,
 		CltvExpiry: 40,
+		Hash:       payHash[:],
 	}
-	ctxt, _ := context.WithTimeout(ctxb, defaultTimeout)
-	carolInvoice, err := carol.AddInvoice(ctxt, invoiceReq)
+	ctxt, cancel := context.WithTimeout(ctxb, defaultTimeout)
+	defer cancel()
+	carolInvoice, err := carol.AddHoldInvoice(ctxt, invoiceReq)
 	if err != nil {
-		t.Fatalf("unable to generate carol invoice: %v", err)
+		t.Fatalf("unable to add invoice: %v", err)
 	}
 
+	// Subscribe to updates on the hodl invoice to find out when it is
+	// marked as accepted.
+	ctx, cancel := context.WithTimeout(ctxb, defaultTimeout)
+	defer cancel()
+	invoiceUpdates, err := carol.SubscribeSingleInvoice(ctx,
+		&lnrpc.PaymentHash{
+			RHash: payHash[:],
+		},
+	)
+
 	// Now that we've created the invoice, we'll send a single payment from
 	// Alice to Carol. We won't wait for the response however, as Carol
 	// will not immediately settle the payment.
-	ctx, cancel := context.WithCancel(ctxb)
-	defer cancel()
-
-	alicePayStream, err := net.Alice.SendPayment(ctx)
-	if err != nil {
-		t.Fatalf("unable to create payment stream for alice: %v", err)
-	}
-	err = alicePayStream.Send(&lnrpc.SendRequest{
-		PaymentRequest: carolInvoice.PaymentRequest,
-	})
-	if err != nil {
-		t.Fatalf("unable to send payment: %v", err)
-	}
+	payErr := make(chan error, 1)
+	go func() {
```

Comment:
Is this change necessary? It emulates the prior behavior, but with an additional goroutine.

Reply:
A similar comment applies to the instance below.

Reply:
Reverted change. It was a left over from a previous different approach to fixing the itests.